src/basic/unit-file.c

```diff
@@ -261,6 +261,86 @@ static int directory_name_is_valid(const char *name) {
         return false;
 }
 
+static int unit_file_resolve_symlink(
+                const char *root_dir,
+                char **search_path,
+                const char *dir,
+                int dirfd,
+                const char *filename,
+                char **ret_simplified,
+                const char **ret_dst) {
+
+        _cleanup_free_ char *target = NULL, *simplified = NULL;
+        const char *dst;
+        int r;
+
+        assert(dir);
+        assert(dirfd >= 0);
+        assert(filename);
+
+        r = readlinkat_malloc(dirfd, filename, &target);
+        if (r < 0)
+                return log_warning_errno(r, "Failed to read symlink %s%s%s: %m",
+                                         dir, dir ? "/" : "", filename);
+
+        const bool is_abs = path_is_absolute(target);
```

Comment:
"const bool"? please let's not start with this. Let's use it on pointers, but on local variables it's pointless clutter really.